### DISCUSSION TITLE
CI testing that requires a large static data store of files

### DISCUSSION THREAD

**Discussion by taldcroft** [community/community#25527] [Actions]
<p>I have around 20 Python package repos where I want to do multiplatform CI testing (linux, mac, windows).  For running tests many of these repos need access to a common set of static data files that total around 2 Gb in size.  Is there a way to efficiently allow access to such a data repository for doing CI tests?</p>
<p>Any help most appreciated!</p>
<ul>
<li>Tom</li>
</ul>

**Comment by weide-zhou** [comment, marked as answer]
<p>You can set the data repo as a submodule of main repo, and checkout the data via use ‘actions/checkout@v1’ with submodule parameter in the workflow run. Please refer to similar <a href="https://github.community/t/how-to-checkout-a-private-submodule-from-a-github-action/16461">ticket</a> for more details.</p>
<p>code sample:</p>
<pre><code>- uses: actions/checkout@v1
  with:
    token: ${{ secrets.ACCESS_TOKEN }}
    submodules: true
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5466cf5e62b58a898f2be55232fc6225967fd903.png" data-download-href="/uploads/short-url/c2EnnbyRKAspAgkLcVguC9PJCGn.png?dl=1" title="submodule.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181091726-39b700b8-9ffa-4996-8a5b-24049b8fccca.png" alt="submodule.png" data-base62-sha1="c2EnnbyRKAspAgkLcVguC9PJCGn" width="690" height="229" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5466cf5e62b58a898f2be55232fc6225967fd903_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">submodule.png</span><span class="informations">955×317 18.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
